prereise/gather/demanddata/transportation_electrification/dwelling.py

```diff
@@ -0,0 +1,98 @@
+def get_segment(dwelling_start_time, total_dwell_period):
+    """Get dwelling activity segment.
+
+    :param float s1: dwelling start time.
+    :param float sn: dwelling end time.
+    :return: (*int*) -- the amount of the rates (cost function) segments the dwelling activity possess.
+    """
+    return total_dwell_period - dwelling_start_time + 1
+
+
+def get_location(location_strategy, dwell_location):
+    """Determines if the vehicle can be charged given location strategy and dwelling location
+
+    :param int location_strategy: where the vehicle can charge-1, 2, 3, 4, 5, or 6;
+        1-home only, 2-home and work related, 3-anywhere if possibile,
+        4-home and school only, 5-home and work and school, 6-only work
+    :param int dwell_location: location the vehicle dwells
+    :return: (*bool*) -- a boolean that represents whether or not the vehicle can charge
+    """
+    location_allowed = {
+        1: dwell_location == 1,
+        2: dwell_location < 13,
+        3: True,
+        4: dwell_location in {1, 21},
+        5: dwell_location in {1, 11, 12, 21},
+        6: (dwell_location >= 10) or (dwell_location < 13),
+    }
+    return location_allowed[location_strategy]
+
+
+def consider_trip_number(trip_strategy, total_trips, trip_num):
+    """Determines if the vehicle should charge given trip strategy and current trip
+
+    :param int trip_strategy: a toggle that determines if should charge on any trip or only
+        after last trip (1-anytrip number, 2-last trip)
+    :param int total_trips: total trips that the vehicle makes
+    :param int trip_num: the trip number of the current trip
+    :return: (*bool*) -- boolean that represents if the vehicle should charge
+    """
+    if trip_strategy == 1:
+        return True
+
+    elif trip_strategy == 2:
+        return total_trips == trip_num
```

Comment:
If `trip_strategy` is neither 1 or 2 the function will return `None`. Is that ok?

Reply:
updated